bundles/automation/org.eclipse.smarthome.automation.provider.file/src/main/java/org/eclipse/smarthome/automation/internal/provider/file/TemplateFileProvider.java

```diff
@@ -53,4 +58,30 @@ public RuleTemplate getTemplate(String UID, Locale locale) {
         return new LinkedList<RuleTemplate>(values);
     }
 
+    protected void setTemplateRegistry(TemplateRegistry<RuleTemplate> templateRegistry) {
+        this.templateRegistry = templateRegistry;
+    }
+
+    protected void removeTemplateRegistry(TemplateRegistry<RuleTemplate> templateRegistry) {
+        this.templateRegistry = null;
+    }
+
+    /**
+     * This method is responsible for checking the existence of {@link ModuleType}s or {@link Template}s with the same
+     * UIDs before these objects to be added in the system.
+     *
+     * @param uid UID of the newly created {@link Template}, which to be checked.
+     * @return {@code true} if {@link Template} with the same UID exists or {@code false} in the opposite
+     *         case.
+     */
+    @Override
+    protected boolean checkExistence(String uid) {
+        if (templateRegistry != null && templateRegistry.get(uid) != null) {
+            logger.error("Rule Template with UID \"{}\" already exists! Failed to create a second with the same UID!",
```

Comment:
Will be moved here:
if (checkExistence(uid)) {
logger.error("Module Type with UID "{}" already exists! Failed to create a second with the same UID!"...
continue;
}